main.go

```diff
@@ -119,11 +133,25 @@ func getPin(pin string) gpio.PinIO {
 	return thePin
 }
 
-func connectMqtt(server string) (MQTT.Client, error) {
+func getButtonPin(pin string) gpio.PinIO {
+	return gpioreg.ByName(pin)
+}
+
+func connectMqtt(config *config) (MQTT.Client, error) {
 	log.Println("mqtt: preparing connection")
+	server := config.mqtt.server
 	connOpts := MQTT.NewClientOptions().AddBroker(server).SetCleanSession(true)
 	tlsConfig := &tls.Config{InsecureSkipVerify: true, ClientAuth: tls.NoClientCert}
 	connOpts.SetTLSConfig(tlsConfig)
+	connOpts.OnConnect = func(client MQTT.Client) {
```

Comment:
ahh now i see why you're passing the entire config....

but why are we subscribing to a topic on connection? is this necessary?

Since this method is `connectMqtt` we should only allow it to return once the connection has been established. Then we should be free to subscribe to the topic of choice.

```go
client, err := connectMqtt(c.mqtt.server)
if err != nil {
  panic("no mqtt available")
}
go watchButton(c.button, client)
...

func watchButton (b configButton, c MQTT.Client) {
  c.Subscribe(b.topic, func (client MQTT.Client, m MQTT.Message) {
    buttonHandler(b.pin)
  }
}
```

Reply:
^ doing this for the same reason as above, we can easily convert the config structure to accept an array of button configs and watch multiple buttons.